Git note research (#1)
packit-service/research: git_notes/README.md

# Git Notes research

The aim of this research is to provide
a feedback around [git notes](https://git-scm.com/docs/git-notes) command.

> [Examples](./examples) directory contains [git_notes.sh](./git_notes.sh).

Reviewer: ```suggestion
```

> The command adds number of your specific commits into your `git` repository.

Reviewer: ```suggestion
There's a [git_notes.sh](./git_notes.sh), which adds number of your specific commits into your `git` repository.
```

Reviewer 2: I'd love to provide suggestion for your suggestion :D 

There's a []() script, which...

Reviewer: suggestion 'commited' :)

## What git_notes.sh does?

> * Add N-notes into your `git repository`.

Reviewer: ```suggestion
* Add N-notes into your git repository.
```

>   * The note added into git by command `git note -m ""`:

Reviewer: ```suggestion
  * The notes are added with `git node add -f -m "Testing notes ${cnt}\npackit_test=yes" -m "json=${JSON_NOTE}" ${commit}:
```

> ```Testing notes ${cnt}\npackit_test=yes\njson={JSON_OBJECT}"```

Reviewer: ```suggestion
```

* Show all notes added by previous command

> * Show specific (3th) note from git.

Reviewer: ```suggestion
* Show specific (2nd) note from git.
```

> ## Can be used git notes as a database?

Reviewer: ```suggestion
## Can git notes be used as a database?
```

`git notes` are basically one liners. We can use a separator and for

> some things it can be used as a "database". The script [git_notes.sh](./git_notes.sh)

Reviewer: ```suggestion
some things it can be used as a "database". The [git_notes.sh](./git_notes.sh) script
```

also stores JSON object into git.
If ``multiple`` `-m` options are given, their values
are concatenated as separate paragraphs.

> ## Can be used git notes for storing states?

Reviewer: ```suggestion
## Can git notes be used for storing states?
```

> Yes, definitely. We can use `git notes` for storing state.

Reviewer: ```suggestion
```

> But, first of all, we have to read note from specific commit and then

Reviewer: ```suggestion
Yes, but, first of all, we have to read note from specific commit and then
```

append the state into git note.
Only one note per one commit is allowed.

For 'editing' git notes, a command
`git note add -f -m "some message"` has to be used.
`-f` option is used for overwriting notes.

## Can we check out the commit and its git notes?

> Yes, `git notes` where designed for it. See [git_notes.sh](./git_notes.sh).

Reviewer: ```suggestion
Yes, `git notes` were designed for it. See [git_notes.sh](./git_notes.sh).
```
